p2p/lightchain.py

```diff
@@ -162,10 +169,14 @@ def peer_block_height(peer: LESPeer) -> int:
                 self._last_processed_announcements[peer] = head_info
             except OperationCancelled:
                 self.logger.debug("Asked to stop, breaking out of run() loop")
+                await self.disconnect_peer(peer, DisconnectReason.client_quitting)
```

Comment:
This is already done in `PeerPool` (although it should send the disconnect reason), as that is where the disconnection will actually take place. Here it is possible we quit the LightChain without stopping the PeerPool, in which case we wouldn't want to disconnect from the peer

Reply:
Ah, ok, I'll take a look at sending the `client_quitting` when `PeerPool` shuts down.

When would we want to quit the `LightPeerChain` but leave the `PeerPool` running?